### PR TITLE
Remove assert_throws use in various remaining tests.

### PR DESCRIPTION
**Pull request by bzbarsky**

### CODE REVIEW

File: common/test-setting-immutable-prototype.js [hunk 1]
```diff
@@ -9,14 +10,21 @@ self.testSettingImmutablePrototypeToNewValueOnly =
...
-    dunderProtoError = new TypeError();
+    // Is this actually right?  Or should this TypeError be thrown in our
+    // global?
+    dunderProtoError = targetGlobal.TypeError;
     dunderProtoErrorName = "TypeError";
   }
 
   test(() => {
-    assert_throws(dunderProtoError, function() {
+    var func = function() {
```
Comment (annevk) [inline common/test-setting-immutable-prototype.js]:
```suggestion
    const func = function() {
```

File: common/test-setting-immutable-prototype.js [hunk 2]
```diff
@@ -9,14 +10,21 @@ self.testSettingImmutablePrototypeToNewValueOnly =
   let dunderProtoError = "SecurityError";
   let dunderProtoErrorName = "\"SecurityError\" DOMException";
   if (isSameOriginDomain) {
-    dunderProtoError = new TypeError();
+    // Is this actually right?  Or should this TypeError be thrown in our
+    // global?
```
Comment (annevk) [inline common/test-setting-immutable-prototype.js]: Both Chrome and Firefox pass the relevant tests it seems, so yes?

**Review by annevk — APPROVED** [review]
Generally looks fine and the couple tests I ran also seem fine. Not sure if we should commit with that comment though?

**Comment by annevk** [comment]
(And I guess there's the problem of the TaskGroup failure. I guess that's due to the number of tests? Would require someone like @jgraham to land.)

**Comment by gsnedders** [comment]
Yeah, let's drop that comment if we know the answer, because otherwise it'll just confuse someone in future.

File: common/test-setting-immutable-prototype.js (hunk 2 above)
Comment (bzbarsky) [reply common/test-setting-immutable-prototype.js]: Ah, sorry, I forgot I left that in.  I just rechecked, and this is in fact right given that `__proto__` is an accessor, so we are entering the Realm of the target here.  I'll remove that comment.

**Review by annevk — APPROVED** [review]

**Review by t10-13rocket — COMMENTED** [review]
LGTM

**Review by cr8t0r — CHANGES_REQUESTED** [review]
this is an unrelated merge

**Pull request merged**